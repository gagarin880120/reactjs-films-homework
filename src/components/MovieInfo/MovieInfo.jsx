import React from 'react';
import styles from './MovieInfo.module.scss';

export default function MovieInfo() {
  return (
    <div className={styles.wrapper}>
      <h2 className={styles.movieTitle}>
        The Jungle Book
      </h2>
      <div className={styles.genreDurationContainer}>
        <span className={styles.genre}>
          Adventure Drama Family Fantasy
        </span>
        <span className={styles.duration}>
          1h 46m
        </span>
      </div>
      <div className={styles.rating}>
          <span className={styles.ratingStars}>
            ★★★★★
          </span>
          <span className={styles.ratingText}>
            4.8
          </span>
        </div>
    </div>
  )
}
