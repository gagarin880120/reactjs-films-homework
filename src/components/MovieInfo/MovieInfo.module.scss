.wrapper {
  width: 100%;
  background-image: url('https://i.ibb.co/7Wh7Jwz/no-backdrop.jpg');
  background-size: cover;
}

.mainContent {
  padding: 0 2% 40px;
  background: linear-gradient(0deg, rgba(0, 0, 0, 0.9) 0%, rgba(0, 0, 0, 0) 100%);
  margin-top: 0%;
  display: flex;
  justify-content: space-between;
  position: relative;
}

.movieDetails {
  width: 40%;
  margin-top: 35%;
}

.movieTitle {
  text-transform: uppercase;
  font-size: 32px;
  font-weight: bold;
}

.genreDurationContainer {
  font-size: 14px;
  width: 100%;
  margin-top: 2%;
}

.genre {
  word-spacing: 15px;
}

.genre::after {
  content: '|';
  margin-left: 20px;
}

.duration {
  margin-left: 5%;
}

.rating {
  margin-top: 5%;
}

.ratingStars {
  color: #09ade3;
  font-size: 26px;
  letter-spacing: 8px;
  vertical-align: middle;
}

.ratingText {
  margin-left: 15px;
  font-size: 18px;
  border: 1px solid #0ab9f3;
  border-radius: 3px;
  padding: 4px 8px;
  background: #020201;
}


@media screen and (max-width: 750px) {
  .mainContent {
    flex-direction: column;
    padding: 2% 0 160px;
  }

  .movieDetails {
    width: 100%;
  }
}
