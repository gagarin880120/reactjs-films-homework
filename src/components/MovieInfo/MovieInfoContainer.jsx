import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MovieInfo from './MovieInfo';
import { movieDetailsSelector } from '../../redux/selectors';

export function MovieInfoContainer({ movie }) {
  return (
    <MovieInfo
      movie={movie}
    />
  );
}

export const mapStateToProps = (state) => ({
  movie: movieDetailsSelector(state),
});

MovieInfoContainer.propTypes = {
  movie: PropTypes.instanceOf(Object),
};

MovieInfoContainer.defaultProps = {
  movie: {},
};

export default connect(mapStateToProps)(MovieInfoContainer);
