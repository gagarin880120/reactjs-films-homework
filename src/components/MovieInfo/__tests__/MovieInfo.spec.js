import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import MovieInfo from '../MovieInfo';

describe('MovieInfo component', () => {
  test('should render without crashing', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<MovieInfo movie={{stars: [{type: 'full', id: 0}]}} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
