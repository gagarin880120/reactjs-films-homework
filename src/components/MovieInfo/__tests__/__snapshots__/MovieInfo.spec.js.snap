// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <h2
    className="movieTitle"
  />
  <div
    className="genreDurationContainer"
  >
    <span
      className="genre"
    />
    <span
      className="duration"
    />
  </div>
  <div
    className="rating"
  >
    <span
      className="ratingStars"
    >
      <RatingStar
        type="full"
      />
    </span>
    <span
      className="ratingText"
    />
  </div>
</div>
`;
